src/components/LinearSchedule.vue

```diff
@@ -46,52 +58,40 @@ export default {
 	},
 	computed: {
 		sessionBuckets () {
-			const buckets = {}
-			for (const session of this.sessions) {
-				const key = session.start.format()
-				if (!buckets[key]) {
-					buckets[key] = []
+			if (this.sortBy == 'time') {
+				const buckets = {}
+				for (const session of this.sessions) {
+					const key = session.start.format()
+					if (!buckets[key]) {
+						buckets[key] = []
+					}
+					if (!session.id) {
+						// Remove duplicate breaks, meaning same start, end and text
+						session.break_id = `${session.start}${session.end}${session.title}`
+						if (buckets[key].filter(s => s.break_id === session.break_id).length === 0) buckets[key].push(session)
+					} else {
+						buckets[key].push(session)
+					}
 				}
-				if (!session.id) {
-					// Remove duplicate breaks, meaning same start, end and text
-					session.break_id = `${session.start}${session.end}${session.title}`
-					if (buckets[key].filter(s => s.break_id === session.break_id).length === 0) buckets[key].push(session)
-				} else {
-					buckets[key].push(session)
-				}
-			}
 
-			return Object.entries(buckets).map(([date, sessions]) => {
-				let sessionBucket = {}
+				return Object.entries(buckets).map(([date, sessions]) => ({
+					date: sessions[0].start,
+					// sort by room for stable sort across time buckets
+					sessions: sessions.sort((a, b) => this.rooms.findIndex(room => room.id === a.room.id) - this.rooms.findIndex(room => room.id === b.room.id))
```

Comment:
This project already include lodash, please use "sort-by-key" strategy (https://lodash.com/docs/4.17.15#sortBy), it produces simpler code.